### PR TITLE
amrex: new version 20.12

### PR DESCRIPTION
**Pull request by mic84**

### CODE REVIEW

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 1]
```diff
@@ -158,7 +161,6 @@ def cmake_args(self):
                                      'linear_solvers'),
             self.define_from_variant('AMReX_AMRDATA', 'amrdata'),
             self.define_from_variant('AMReX_PARTICLES', 'particles'),
-            self.define_from_variant('AMReX_SUNDIALS', 'sundials'),
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/amrex/package.py:161]: This doesn't seem to be conditional on versions. If that's correct we should either completely remove the variant or have something like:
```python
if self.spec.satisfies('@:20.11'):
     args.append(self.define_from_variant('AMReX_SUNDIALS', 'sundials'))
```

**Review by alalazo — CHANGES_REQUESTED** [review]
See comment below.

File: var/spack/repos/builtin/packages/amrex/package.py (hunk 1 above)
Comment (mic84) [reply var/spack/repos/builtin/packages/amrex/package.py:161]: I provided two versions of `cmake_args` depending on the amrex version to build. `AMReX_SUNDIALS` was removed from the version of `cmake-args` used for >= 20.11.  This should take care of the conditionality.

Comment (alalazo) [reply var/spack/repos/builtin/packages/amrex/package.py:161]: Thanks, it wasn't evident from Github's diff but obviously you're correct.

**Review by alalazo — APPROVED** [review]

**Pull request merged**